pbsmetrics/prometheus/prometheus.go

```diff
@@ -387,6 +427,36 @@ func (m *Metrics) RecordRequestTime(labels pbsmetrics.Labels, length time.Durati
 	}
 }
 
+func (m *Metrics) RecordStoredDataFetchTime(labels pbsmetrics.StoredDataTypeLabels, length time.Duration) {
```

Comment:
One of the labels passed in here is `DataType` but that is just used to figure out which timer to update; it isn't an actual prometheus label. I figured maybe that shouldn't be a label and instead I should just pass a `StoredDataType` parameter to this function and `switch` on that. Thoughts?

Reply:
Perhaps we should have just one timer and let that be a label as well. I don't think it increases cardinality by breaking the metrics off into the various `DataType`s, and it would allow graphing the overall performance over all the datatypes. What benefit do we get by breaking them into distinct metrics?

Reply:
That's how I originally had it 🙂. @SyntaxNode and I chatted and we figured it would be too many data points for one timer such that it might affect performance. (12 buckets + 1 count + 1 sum) * 5 data types * 2 fetch types * 2 fetch statuses = 280.
@SyntaxNode anything else to add?

Reply:
This was addressed offline. We decided to keep separate timers for each stored data type for performance reasons but to create an additional set of stored data counters for errors.